com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl

```diff
@@ -1315,6 +1315,20 @@ real UnpackHeightmap(real4 height)
 // Misc utilities
 // ----------------------------------------------------------------------------
 
+// Decodes HDR textures. Handles full HDR, dLDR and RGBM formats
+float3 DecodeHDR(float4 data, float4 decodeInstructions)
```

Comment:
> note that your function is very close to this one in EntityLighting:
> 
> real3 DecodeHDREnvironment(real4 encodedIrradiance, real4 decodeInstructions) { // Take into account texture alpha if decodeInstructions.w is true(the alpha value affects the RGB channels) real alpha = max(decodeInstructions.w * (encodedIrradiance.a - 1.0) + 1.0, 0.0);
> 
> ```
> // If Linear mode is not supported we can skip exponent part
> return (decodeInstructions.x * PositivePow(alpha, decodeInstructions.y)) * encodedIrradiance.rgb;
> ```
> 
> }
> 
> so try to update this function instead?

Damn, I looked around for a function like that already present but overlooked this file. This function will suit our needs as is, since HDRP always uses Linear color space. I'll switch to use it.